src/databricks/labs/ucx/hive_metastore/locations.py

```diff
@@ -248,3 +251,80 @@ def _try_fetch(self) -> Iterable[Mount]:
             f"SELECT * FROM {escape_sql_identifier(self._schema)}.{escape_sql_identifier(self._table)}"
         ):
             yield Mount(*row)
+
+
+class TablesInMounts(CrawlerBase[Table]):
+
+    def __init__(
+        self,
+        backend: SqlBackend,
+        ws: WorkspaceClient,
+        inventory_database: str,
+        mc: Mounts,
+        include_mounts: list[str] | None = None,
+    ):
+        super().__init__(backend, "hive_metastore", inventory_database, "tables", Table)
+        self._dbutils = ws.dbutils
+        self._mc = mc
+        self._include_mounts = include_mounts
+
+    def snapshot(self) -> list[Table]:
+        """
+        Takes a snapshot of tables in the specified catalog and database.
+
+        Returns:
+            list[Table]: A list of Table objects representing the snapshot of tables.
+        """
+        return self._snapshot(partial(self._try_load), partial(self._crawl))
+
+    def _try_load(self) -> Iterable[Table]:
+        """Tries to load table information from the database or throws TABLE_OR_VIEW_NOT_FOUND error"""
+        for row in self._fetch(f"SELECT * FROM {escape_sql_identifier(self.full_name)}"):
+            yield Table(*row)
+
+    def _crawl(self):
+        all_mounts = self._mc.snapshot()
+        all_tables = []
+        for mount in all_mounts:
+            if self._include_mounts and mount.name not in self._include_mounts:
+                continue
+            table_paths = self._find_delta_log_folders(mount.name)
+            for path in table_paths:
+                table = Table(
+                    catalog="hive_metastore",
+                    database="",
```

Comment:
Database can't be empty. Infer it from path and default to "default"

Reply:
I don't really want to do the mapping of paths -> databases/tables there. As this will be handled later in the process. 
What if instead just using the mount name as a database ?

Reply:
Actually, it would even better to default the database of this to 'mounts' or something that would be easily searcheable for downstream applications that must use this table.

Reply:
and what if there are duplicate table names?...

Reply:
yeah that can totally happen ... the strategy of picking the parent folder as the table name is irrelevant then. 
Maybe let's simply use the table path in mount as the table name instead ? ...